R/ui-legacy.R

```diff
@@ -248,25 +257,14 @@ ui_value <- function(x) {
   x
 }
 
-#' @rdname ui
+#' @rdname ui-legacy-functions
 #' @export
 #' @param base If specified, paths will be displayed relative to this path.
 ui_path <- function(x, base = NULL) {
-  is_directory <- is_dir(x) | grepl("/$", x)
-  if (is.null(base)) {
-    x <- proj_rel_path(x)
-  } else if (!identical(base, NA)) {
-    x <- path_rel(x, base)
-  }
-
-  # rationalize trailing slashes
-  x <- path_tidy(x)
-  x <- ifelse(is_directory, paste0(x, "/"), x)
-
-  ui_value(x)
+  ui_value(ui_path_impl(x, base = base))
```

Comment:
`ui_path()` does 2 things:

* "path math", i.e. possibly making the path relative to some base, probably the active project's root
* styling as a value

I still need the "path math", so that is now implemented in `ui_path_impl()`, which is aliased to the shorter form `pth()`.